index.js

```diff
@@ -672,65 +655,56 @@ class GitlabScm extends Scm {
-            json: true,
-            method: 'GET',
-            auth: {
-                bearer: token
-            },
-            url:
-                `${this.config.gitlabProtocol}://${this.config.gitlabHost}/api/v4` +
-                `/projects/${owner}%2F${reponame}` +
-                `/repository/commits/${sha}`
-        });
-
-        checkResponseError(commit, '_decorateCommit: commitLookup');
-
-        const authorName = Hoek.reach(commit, 'body.author_name');
-        const committerName = Hoek.reach(commit, 'body.committer_name');
-        const author = { ...DEFAULT_AUTHOR };
-        const committer = { ...DEFAULT_AUTHOR };
-
-        if (authorName) {
-            author.name = authorName;
-        }
+        return this.breaker
+            .runCommand({
+                method: 'GET',
+                token,
+                route: `projects/${owner}%2F${reponame}/repository/commits/${sha}`,
+                caller: '_decorateCommit: commitLookup'
+            })
+            .then(commit => {
+                const authorName = Hoek.reach(commit, 'body.author_name');
+                const committerName = Hoek.reach(commit, 'body.committer_name');
+                const author = { ...DEFAULT_AUTHOR };
+                const committer = { ...DEFAULT_AUTHOR };
+
+                if (authorName) {
+                    author.name = authorName;
+                }
 
-        if (committerName) {
-            committer.name = committerName;
-        }
+                if (committerName) {
+                    committer.name = committerName;
+                }
 
-        return {
-            author,
-            committer,
-            message: commit.body.message,
-            url: commit.body.web_url
-        };
+                return {
+                    author,
+                    committer,
+                    message: commit.body.message,
+                    url: commit.body.web_url
+                };
+            });
     }
 
     /**
-     * Decorate the author based on the Gitlab service
+     * Decorate the author based on the GitLab service
      * @async _decorateAuthor
      * @param  {Object}        config            Configuration object
-     * @param  {Object}        config.token      Service token to authenticate with Gitlab
+     * @param  {Object}        config.token      Service token to authenticate with GitLab
      * @param  {Object}        config.username   Username to query more information for
      * @return {Promise}
      */
     async _decorateAuthor({ token, username }) {
         return this.breaker
             .runCommand({
-                json: true,
                 method: 'GET',
-                auth: {
-                    bearer: token
-                },
-                url: `${this.config.gitlabProtocol}://${this.config.gitlabHost}/api/v4/users`,
-                qs: {
+                token,
+                route: `users`,
+                json: {
```

Comment:
how come `json` is being passed for `GET` request ? 

Also for we seem to be using https://github.com/sindresorhus/got/blob/main/documentation/2-options.md#allowgetbody , why ?

Reply:
I thought we needed to pass `qs` as `json` but it should be `searchParams`
fixed: https://github.com/screwdriver-cd/request/pull/4